### PR TITLE
Update SDK

### PR DESCRIPTION
**Pull request by BrennanConroy**
Hopefully this will have some changes for the ARM issue we're seeing in https://github.com/dotnet/aspnetcore/issues/24841

### CODE REVIEW

**Comment by pranavkm** [comment]
💥

**Review by dougbu — CHANGES_REQUESTED** [review]
Looks like the latest SDK still has problems. Or, at least it can't work on the agents we're using.

/cc @dsplaisted @marcpopMSFT @sfoslund @clairernovotny @nkolev92

**Comment by dougbu** [comment]
Put another way, https://github.com/dotnet/sdk/pull/12897 didn't help us.

**Comment by BrennanConroy** [comment]
> Put another way, [dotnet/sdk#12897](https://github.com/dotnet/sdk/pull/12897) didn't help us.

Are we sure that change is in this version we're grabbing?

**Comment by dougbu** [comment]
Not certain though the change was merged 4 days ago and the SDK used here is from this morning.

**Comment by marcpopMSFT** [comment]
What version of VS is being used here? The problem may be that you need 16.8.p2 to work with this sdk.  @nkolev92 , does that sound right?  What is their path forward until that's available?

**Comment by BrennanConroy** [comment]
> What version of VS is being used here?

Shouldn't matter for the Unix machines that are failing with the same error.

**Comment by dougbu** [comment]
> What version of VS is being used here?

For example, from the macOS build:
``` text
/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20420.10/NuGet.targets(128,5): error : Invalid restore input. Invalid target framework 'unsupported'. Input files: /Users/runner/work/1/s/.packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.20417.6/tools/Tools.proj. [/Users/runner/work/1/s/.packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.20417.6/tools/Tools.proj]
```

It appears the .NET SDK contains a breaking change that required the https://github.com/dotnet/arcade/commit/f8d13313debbaf9777dba6c3059103c0cd303636#diff-49032945af4968a76e1ae76f37234d19 workaround. We need a new Arcade SDK to get things working with a newer .NET SDK.

@BrennanConroy I suggest triggering our Arcade subscription because they just promoted a new build. According to https://github.com/dotnet/arcade/compare/83fda4b3f6d93e713749fd1b27c4a6d40b118b13...56a95cc477558c1ccdf16d7abe962849ea970ba4, the build includes the above workaround.

**Review by dougbu — APPROVED** [review]
Withdrawing my changes request in hopes this works.

nit: I meant `trigger-subscriptions` not `update-dependencies` but am fine w/ keeping everything in this PR.

**Comment by BrennanConroy** [comment]
Looks like some extra checks have been added. I'll fix the taskcontinuationoptions, not sure about netcoreapp3.0 yet.

**Comment by dougbu** [comment]
Oops, new errors about targeting `netcoreapp3.0` and a breaking change somewhere. But, looks better than the last run❕

**Comment by BrennanConroy** [comment]
This is going to be a fun PR 😰

File: src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ public async ValueTask<TValue> InvokeAsync<TValue>(string identifier, object[] a
         public ValueTask<TValue> InvokeAsync<TValue>(string identifier, CancellationToken cancellationToken, object[] args)
         {
             var taskId = Interlocked.Increment(ref _nextPendingTaskId);
-            var tcs = new TaskCompletionSource<TValue>(TaskContinuationOptions.RunContinuationsAsynchronously);
+            var tcs = new TaskCompletionSource<TValue>(TaskCreationOptions.RunContinuationsAsynchronously);
```
Comment (BrennanConroy) [inline src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs:91]: Looks like a bunch of tests relied on this being inline instead of async

**Comment by BrennanConroy** [comment]
>     1. This PR still contains an Arcade SDK update. Should it❔

That's just because git isn't showing the PR rebased on release/5.0 correctly since the arcade update went in from a different PR.

>     2\. Should this PR target release/5.0 at this point in the release❔

Why not? Unless we're planning on perpetually targeting an old SDK in 5.0, then we'll need to update it at some point and the earlier the better.

File: src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs (hunk 1 above)
Comment (wtgodbe) [reply src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs:91]: ping @pranavkm @captainsafia can we get this resolved today?

**Comment by dougbu** [comment]
> Why not? Unless we're planning on indefinitely targeting an old SDK in 5.0, then we'll need to update it at some point and the earlier the better.

Because we're in ask mode, the change is unreasonably painful, and moving to something other than (say) the stable / released RC1 SDK after that's out seems nice-to-have but not required.

**Comment by wtgodbe** [comment]
We could go for the preview8 SDK once it releases tomorrow

**Comment by wtgodbe** [comment]
We're going to punt this work until we get the RC1 SDK

**Pull request closed without merging**